src/modules/client/pages/ni/customers/CustomersDirectory.vue

```diff
@@ -209,7 +209,7 @@ export default {
         const isValid = await this.waitForFormValidation(this.$v.newCustomer);
         if (!isValid) return NotifyWarning('Champ(s) invalide(s)');
 
-        const payload = pickBy(this.newCustomer);
+        const payload = pickBy(this.newCustomer); // à quoi ça sert ?
```

Comment:
Je ne vois pas ce que pickBy sans prédicat fait, à part créer un nouvelle objet pour éviter de muter this.newCustomer mais je ne vois pas pourquoi this.newCustomer serait muté par Customer.create().
https://lodash.com/docs/4.17.15#pickBy

Reply:
ça permet d'avoir le format attendu par l'appel du back non ?

Reply:
Je ne vois vraiment pas pourquoi, pickBy est censé te sélectionner des champs selon une fonction et te renvoyer un objet composé de ces champs.
Je ne vois pas pourquoi on l'utilises ici 🤔

Reply:
@sophiemoustard @cmolle une idée ?

Reply:
sans 2eme parametre, le pickBy permet de supprimer un champ vide
`pickBy({ customer: 'poiuytrty', subscription: '' })` -> `{ customer: 'poiuytrty' }`

Reply:
mais ici il ne sert a rien car le pickBy ne regarde que les champs a la racine de l'objet et ne va pas en profondeur donc je pense que tu peux l'enlever... a tester bien sur :)

Reply:
Je l'ai supprimé ! J'ai dû supprimer le champ email qui n'était pas utilisé, c'est lui qui cassait la validation back puisque la route n'attend pas de champ email.